### ISSUE TITLE
bug with recent v6.21.4 minified aliases in ORDER BY clause

### ISSUE THREAD

**Issue by bminer** [sequelize/sequelize#15422]
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description

v6.21.4 causes a bug (not present in v6.21.3) when generating the `ORDER BY` clause.

v6.21.3 is correct:

```sql
ORDER BY `order`.`ship_to_company` ASC, `order`.`ship_to_country` ASC, `order`.`ship_to_state` ASC, `order`.`ship_to_city` ASC, `order`.`ship_to_zip` ASC, `order`.`ship_to_address` ASC, `ServiceOrderLines`.`id` ASC;
```

v6.21.4 is incorrect:

```sql
ORDER BY `orderID``ship_to_company` ASC, `orderID``ship_to_country` ASC, `orderID``ship_to_state` ASC, `orderID``ship_to_city` ASC, `orderID``ship_to_zip` ASC, `orderID``ship_to_address` ASC, `ServiceOrderLines`.`id` ASC;
```

I have a table called `orders`, and when joining its alias becomes `order`:

```sql
LEFT OUTER JOIN `orders` AS `order` ON `ServiceOrderLines`.`order` = `order`.`id`
```

### Reproducible Example

I will try to use SSCCE shortly, but I want to get this issue submitted, since I believe this is a trivial bugfix.

### What do you expect to happen?

Produce valid SQL

### What is actually happening?

Invalid ORDER BY clause is generated, and the code is re-writing `order` table alias as `orderID` for some reason.

### Environment

- Sequelize version: v6.21.4
- Node.js version: v16.18.1
- Database & Version: MySQL 8
- Connector library & Version: mysql2@2.3.3

## Would you be willing to resolve this issue by submitting a Pull Request?

Yes, but I'd like to get your feedback first.

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in resolving my issue.

---

*Indicate your interest in the resolution of this issue by adding the 👍 reaction. Comments such as "+1" will be removed.*

**Comment by bminer** [comment]
Culprit commit is likely 5a257bc93c7e760f6b0158f55b3cb48878698450.  See PR #14804.

**Comment by WikiRik** [comment]
@evanrittenhouse I haven't read it completely but I think this might be something you can look into since you worked on the other issues as well

**Comment by ephys** [comment]
I have tried to replicate something that could produce this error bug couldn't. I see you mentioned you were working on a SSCCE. Don't hesitate to ping us when you have one so we can debug this further :)

**Comment by evanrittenhouse** [comment]
I also can't reproduce - waiting on SSCCE.

**Comment by bminer** [comment]
I'll try to get the SSCCE out at some point soon.

The problem happens when using the `order` as a multi-element array as [documented here](https://sequelize.org/docs/v6/core-concepts/model-querying-basics/#ordering).

For example, the following query fails when `ServiceOrderLines` has an `order` field with Sequelize alias `orderID`.  In other words, `models.ServiceOrderLines.fieldAttributeMap.order` is "orderID".

```js
ServiceOrderLines.findAll({
  include: [ { association: 'order' } ],
  order: [ [ 'order', 'ship_to_company', 'ASC' ] ] // here 'order' refers to the association, not the field
})
```

In the above case, the ORDER BY clause looks like this:

```sql
ORDER BY `orderID``ship_to_company` ASC
```

But, the following similar query works fine when `orderID` is not among the attributes requested:

```js
ServiceOrderLines.findAll({
  attributes: ["id"],
  include: [{ association: 'order' }],
  order: [['order', 'ship_to_company', 'ASC']]
})
```

**Comment by bminer** [comment]
SSCCE is here: https://github.com/sequelize/sequelize-sscce/pull/243

I hope I did it correctly!

**Comment by ephys** [comment]
You did, thank you :)

**Comment by bminer** [comment]
Any thoughts on this? It is preventing me from upgrading to latest sequelize version. Any way I can help?

**Comment by evanrittenhouse** [comment]
@bminer work has been crazy recently - I may not get to this for a while still. Feel free to give it a shot if you'd like!

**Comment by bminer** [comment]
@sdepold - Could you please take a quick gander at this? I think the bug was created by 5a257bc93c7e760f6b0158f55b3cb48878698450. Can you briefly explain what this change was designed to do?

The bug is [succinctly explained here](https://github.com/sequelize/sequelize/issues/15422#issuecomment-1351921184)

I'm afraid that if I write a PR for this, I'll just break something.

**Comment by WikiRik** [comment]
@bminer that change was a backport for https://github.com/sequelize/sequelize/pull/14804 which fixed the issues mentioned in https://github.com/sequelize/sequelize/issues/14723. Does that give enough information?

Does this bug occur on the latest v7 alpha as well? Because in that case we'll have to fix it in both, starting with `main`

**Comment by bminer** [comment]
I can confirm that 7.0.0-alpha.9 is working properly, but 6.32.1 still has the bug.

**Comment by bminer** [comment]
Okay, after testing the **latest** 7.0.0-alpha.27, the bug is present.

**Comment by windy-itus** [comment]
I still have this problem on 6.33.0.
Please let me know the version that solves this issue

**Comment by WikiRik** [comment]
There is no version yet that solves this issue. But feel free to make a PR that solves this issue on the `main` branch

**Comment by bminer** [comment]
So after looking into this a bit more, it appears that the issue is in the `abstract/query-generator.js` in the `getQueryOrders` function. Namely, there is some code there that relies on `order[0]` matching the field name, but it does not handle the case when the `order` is 3 elements like this: `["collection", "field", "DESC"]` instead of just 2 elements `["field", "DESC"]`.

**Comment by WikiRik** [comment]
Thanks for looking into this more! Can you make a PR for `main` with a new test? After we've merged that, I should be okay with backporting that fix to v6 (the PR you've opened already) and merge that without an additional test for v6

**Comment by bminer** [comment]
I don't think this is a bug anymore in v7 alpha. I cannot seem to replicate it.

**Comment by bminer** [comment]
Any chance we can get this fixed for v6?
